src/sentry/static/sentry/app/views/issueList/createSavedSearchButton.tsx

```diff
@@ -1,171 +1,48 @@
-    this.setState({query: String(val)});
-  };
-
-  render() {
-    const {isSaving, isModalOpen, error} = this.state;
-    const {organization, query, buttonClassName, iconOnly, withTooltip} = this.props;
-
-    return (
-      <Access organization={organization} access={['org:write']}>
-        <Button
-          title={withTooltip ? t('Add to organization saved searches') : undefined}
-          onClick={this.onToggle}
-          data-test-id="save-current-search"
-          size="zero"
-          borderless
-          type="button"
-          aria-label={t('Add to organization saved searches')}
-          icon={<IconAdd size="xs" />}
-          className={buttonClassName}
-        >
-          {!iconOnly && t('Create Saved Search')}
-        </Button>
-        <Modal show={isModalOpen} animation={false} onHide={this.onToggle}>
-          <form onSubmit={this.onSubmit}>
-            <Modal.Header>
-              <h4>{t('Save Current Search')}</h4>
-            </Modal.Header>
-
-            <Modal.Body>
-              {this.state.error && (
-                <div className="alert alert-error alert-block">{error}</div>
-              )}
-
-              <p>{t('All team members will now have access to this search.')}</p>
-              <TextField
-                key="name"
-                name="name"
-                label={t('Name')}
-                placeholder="e.g. My Search Results"
-                required
-                onChange={this.handleChangeName}
-              />
-              <TextField
-                key="query"
-                name="query"
-                label={t('Query')}
-                value={query}
-                required
-                onChange={this.handleChangeQuery}
-              />
-            </Modal.Body>
-            <Modal.Footer>
-              <Button
-                priority="default"
-                size="small"
-                disabled={isSaving}
-                onClick={this.onToggle}
-                style={{marginRight: space(1)}}
-              >
-                {t('Cancel')}
-              </Button>
-              <Button priority="primary" size="small" disabled={isSaving}>
-                {t('Save')}
-              </Button>
-            </Modal.Footer>
-          </form>
-        </Modal>
-      </Access>
-    );
-  }
-}
-
-export default withApi(CreateSavedSearchButton);
+const CreateSavedSearchButton = ({
+  buttonClassName,
+  withTooltip,
+  iconOnly,
+  organization,
+  ...rest
+}: Props) => (
+  <Access organization={organization} access={['org:write']}>
+    <Button
+      title={withTooltip ? t('Add to organization saved searches') : undefined}
+      onClick={() =>
+        openModal(deps => (
+          <CreateSavedSearchModal organization={organization} {...rest} {...deps} />
+        ))
+      }
+      data-test-id="save-current-search"
+      size="zero"
+      borderless
+      type="button"
+      aria-label={t('Add to organization saved searches')}
```

Comment:
I believe there is a `label` prop on the Button, but it will work like this too.